Animation Docs (#759)
lambdaclass/curse_of_myrra: docs/src/skill.md

# Skill.cs Animations

> Lets start by explaining the differents methods we have in this script, since this script handle more things than just animations, lets focus only in the methods and flow of the animations.

Reviewer: *handles

> Lets start by explaining the differents methods we have in this script, since this script handles more things than just animations, lets focus only in the methods and flow of the animations.

Reviewer: ```suggestion
Let's start by explaining the different methods we have in this script, since it handles more things than just animations, let's focus only in the methods and flow of the animations.
```

`SetSkill` 

> It is use to initialize the skills once the character is selected,  it maps the skill from the backend, the skill scripteableObject, the animationEvent.

Reviewer: *It is used. 
is it scripteableObject or ScriptableObject (no e after t)? Check it because I saw it written both ways throughout different texts

Author: Yup it's ScriptableObject , I spelled wrong in this case

> It is used to initialize the skills once the character is selected,  it maps the skill from the backend, the skill ScriptableObject, the animationEvent.

Reviewer: ```suggestion
It is used to initialize the skills once the character is selected, it maps the skill from the backend, the skill ScriptableObject and the animationEvent.
```

`ClearAnimator` 

> This method clear each skill parameter, setting all booleans to false.

Reviewer: *clears

`ChangeCharacterState`

Updates the current playing animation, changes the movement state machine to attacking and sets the respective animation parameter to `true`.

`StartFeedback`

> This is in charge of begin the start animations (parameters with "_start" in their name, ex: "Skill1_start"), calls the methods mencioned before `ClearAnimator` and `ChangeCharacterState` and finally starts a coroutine to end the animation depending of the `startAnimationDuration` time of the skill scripteableObject.

Reviewer: This is in charge of beginning.
mentioned

`ExecuteFeedback`

> #### This method is very importart due its the one which controls all the flow of the Skill animations.

Reviewer: because instead of due

> #### This method is very importart because its the one which controls all the flow of the Skill animations.

Reviewer: ```suggestion
#### This method is very important because it controls all the flow of the Skill animations.
```

> This method implements the same logic as `StartFeedback` but for the parameters of all the skills without the "_start" in their name (ex: Skill1) and also use `executeAnimationDuration` instead of the `startAnimationDuration` to end the animation.

Reviewer: ```suggestion
This method implements the same logic as `StartFeedback` but for the parameters of all the skills without the "_start" in their name (ex: Skill1). It also uses `executeAnimationDuration` instead of `startAnimationDuration` to end the animation.
```

`EndSkillFeedback`

> Changes the movement machine state to Idle and set the animation parameter to false. This is used in the `SkillAnimationEvents` to end the animations.

Reviewer: ```suggestion
Changes the movement machine state to Idle and sets the animation parameter to false. This is used in the `SkillAnimationEvents` to end the animations.
```

## ExecuteFeedback flowchart

![](./images/executeFeedback.png)

> ### This flowchart does only have in count the animation part of the method

Reviewer: ```suggestion
### This flowchart only takes into account the animation part of the method
```

# Skill.cs Animations

Lets start by explaining the differents methods we have in this script, since this script handles more things than just animations, lets focus only in the methods and flow of the animations.

`SetSkill` 

> It is used to initialize the skills once the character is selected,  it maps the skill from the backend, the skill ScripteableObject, the animationEvent.

Reviewer: ScriptableObject

`ClearAnimator` 

This method clears each skill parameter, setting all booleans to false.

`ChangeCharacterState`

Updates the current playing animation, changes the movement state machine to attacking and sets the respective animation parameter to `true`.

`StartFeedback`

> This is in charge of beginning the start animations (parameters with "_start" in their name, ex: "Skill1_start"), calls the methods mentioned before `ClearAnimator` and `ChangeCharacterState` and finally starts a coroutine to end the animation depending of the `startAnimationDuration` time of the skill scripteableObject.

Reviewer: ScriptableObject

Author: Awesome catches!
